### ISSUE TITLE
Incompatibility with numpy 1.22

### ISSUE THREAD

**Issue by yetinam** [obspy/obspy#2912]
Obspy (1.2.2, installed via pip) seems to be incompatible with numpy 1.22 (which is automatically installed through pip when installing obspy). When importing obspy, an AttributeError occurs. I pasted the full error log below. The problem does not occur when using numpy 1.21. The problem occured on Ubuntu, MacOS and Windows in python 3.8 and 3.9.

The issue occurs here: https://github.com/obspy/obspy/blob/c4faef6b2df110e844c28ca2aa9515888c019cb8/obspy/core/util/misc.py#L205-L215
From what I can tell, this code block checks whether the `numpy.loadtxt` function accepts a parameter `ndmin`. However, this parameter has been introduced in numpy 1.6.0, while obspy requires at numpy>=1.15.0. Therefore, I'd suggest to simply remove the function, which should also resolve the version incompatibility issue.

```
In [1]: import obspy
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-1-42a9ad3b2ac1> in <module>
----> 1 import obspy

~/anaconda3/envs/tmp_version_tests/lib/python3.9/site-packages/obspy/__init__.py in <module>
     37 
     38 # don't change order
---> 39 from obspy.core.utcdatetime import UTCDateTime  # NOQA
     40 from obspy.core.util import _get_version_string
     41 __version__ = _get_version_string(abbrev=10)

~/anaconda3/envs/tmp_version_tests/lib/python3.9/site-packages/obspy/core/__init__.py in <module>
    122 
    123 # don't change order
--> 124 from obspy.core.utcdatetime import UTCDateTime  # NOQA
    125 from obspy.core.util.attribdict import AttribDict  # NOQA
    126 from obspy.core.trace import Stats, Trace  # NOQA

~/anaconda3/envs/tmp_version_tests/lib/python3.9/site-packages/obspy/core/utcdatetime.py in <module>
     25 import numpy as np
     26 from obspy.core.compatibility import py3_round
---> 27 from obspy.core.util.deprecation_helpers import ObsPyDeprecationWarning
     28 
     29 # Regular expression used in the init function of the UTCDateTime objects which

~/anaconda3/envs/tmp_version_tests/lib/python3.9/site-packages/obspy/core/util/__init__.py in <module>
     25 # import order matters - NamedTemporaryFile must be one of the first!
     26 from obspy.core.util.attribdict import AttribDict
---> 27 from obspy.core.util.base import (ALL_MODULES, DEFAULT_MODULES,
     28                                   NATIVE_BYTEORDER, NETWORK_MODULES,
     29                                   NamedTemporaryFile, _read_from_plugin,

~/anaconda3/envs/tmp_version_tests/lib/python3.9/site-packages/obspy/core/util/base.py in <module>
     34 from pkg_resources import get_entry_info, iter_entry_points
     35 
---> 36 from obspy.core.util.misc import to_int_or_zero, buffered_load_entry_point
     37 
     38 

~/anaconda3/envs/tmp_version_tests/lib/python3.9/site-packages/obspy/core/util/misc.py in <module>
    212 try:
    213     from numpy import loadtxt
--> 214     loadtxt(np.array([0]), ndmin=1)
    215 except TypeError:
    216     # otherwise redefine loadtxt

~/anaconda3/envs/tmp_version_tests/lib/python3.9/site-packages/numpy/lib/npyio.py in loadtxt(fname, dtype, comments, delimiter, converters, skiprows, usecols, unpack, ndmin, encoding, max_rows, like)
   1084         # the need for decoding and estimate the number of columns.
   1085         for first_line in line_iter:
-> 1086             ncols = len(usecols or split_line(first_line))
   1087             if ncols:
   1088                 # Put first_line back.

~/anaconda3/envs/tmp_version_tests/lib/python3.9/site-packages/numpy/lib/npyio.py in split_line(line)
    975         """Chop off comments, strip, and split at delimiter."""
    976         for comment in comments:  # Much faster than using a single regex.
--> 977             line = line.split(comment, 1)[0]
    978         line = line.strip('\r\n')
    979         return line.split(delimiter) if line else []

AttributeError: 'numpy.int64' object has no attribute 'split'
```

**Comment by d-chambers** [comment]
Hi @yetinam,

Good suggestion. Mind submitting a PR?

**Comment by yetinam** [comment]
Sure, I should get to it in the next days.

**Comment by Molkree** [comment]
Now that the fix for this issue was merged into the 1.2 maintenance branch, are there any plans to release a new patch version?
